### PR TITLE
Add internal utilities.

### PR DESCRIPTION
**Pull request by nevinheintze**
Add package internal, which implements database-agnostic functionality.

### CODE REVIEW

File: internal/progress_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProgress(t *testing.T) {
+	var total int64 = 4321
```
Comment (larkee) [inline internal/progress_test.go]: ``total := int64(4321)``

File: internal/progress_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	p := NewProgress(total, "Progress", false)
+	assert.Equal(t, 0, p.pct)
+	time.Sleep(500 * time.Millisecond)
+	for _, i := range []int64{1000, 2000, 3000} {
```
Comment (larkee) [inline internal/progress_test.go]: Prefer ``v`` over ``i`` to avoid confusion over index vs value.

File: internal/progress_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		p.MaybeReport(i)
+		assert.Equal(t, int((100*i)/total), p.pct)
+		time.Sleep(500 * time.Millisecond)
+	}
+	pct := p.pct
+	p.MaybeReport(2000)
+	assert.Equal(t, pct, p.pct) // pct is monotonic.
+	p.MaybeReport(5000)
+	assert.Equal(t, 100, p.pct) // Never exceed 100%.
+	p.Done()
```
Comment (larkee) [inline internal/progress_test.go]: Should this line be here? Are you missing a subsequent check?

File: internal/progress_test.go (hunk 1 above)
Comment (nevinheintze) [reply internal/progress_test.go]: Done.
Sorry, I missed this - thanks!

File: internal/progress_test.go (hunk 2 above)
Comment (nevinheintze) [reply internal/progress_test.go]: Done.

File: internal/progress_test.go (hunk 3 above)
Comment (nevinheintze) [reply internal/progress_test.go]: Got rid of this, and added a separate test for Done.

**Pull request merged**